pkgs/by-name/ed/edopro/package.nix

```diff
@@ -0,0 +1,264 @@
+    makeFlags = "-C source/Irrlicht";
+
+    installPhase = ''
+      mkdir -p $out/{bin,include}
+      cp lib/Linux/libIrrlicht.a $out/bin
+      cp -r include/* $out/include
+    '';
+  };
+
+  ocgcore = stdenv.mkDerivation rec {
+    pname = "ygopro-core";
+    # Tied to the tagged version of EDOPro's git module.
+    version = "4a872ad7686e0d00ee4c1c051e90fc07c40613d4";
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = pname;
+      rev = version;
+      hash = "sha256-2E1cjG0FONu/fbr67/3qRCKQ1W/wPznClEWsMa1FAzo=";
+      fetchSubmodules = true;
+    };
+
+    nativeBuildInputs = [ premake5 ];
+
+    buildInputs = [
+      lua5_3
+    ];
+
+    preBuild = ''
+      premake5 gmake2
+    '';
+
+    enableParallelBuilding = true;
+    buildFlags = "verbose=true config=release_x64 ocgcore";
+    makeFlags = "-C build";
+
+    installPhase = ''
+      mkdir -p $out/bin
+      cp bin/x64/release/libocgcore.a $out/bin
+    '';
+  };
+
+  edopro = stdenv.mkDerivation rec {
+    pname = "edopro";
+    version = edopro-version;
+
+    src = fetchFromGitHub {
+      owner = "edo9300";
+      repo = pname;
+      rev = "c713e23491a1e55c9d8e91257e5f2b5873696b9b";
+      hash = "sha256-2E1cjG0FONu/fbr67/3qRCKQ1W/wPznClEWsMa1FAzo=";
+      fetchSubmodules = true;
+    };
+
+    nativeBuildInputs = [ premake5 ];
+
+    buildInputs = [
+      bzip2
+      curl
+      flac
+      fmt_9
+      freetype
+      irrlicht-edopro
+      libevent
+      libgit2
+      libjpeg
+      libpng
+      libvorbis
+      mono
+      nlohmann_json
+      openal
+      SDL2
+      sqlite
+    ];
+
+    # nixpkgs' gcc stack currently appears to not support LTO
+    postPatch = ''
+      sed -i '/LinkTimeOptimization/d' ./premake5.lua
+    '';
+
+    # TODO sound, can't get it to work with either SDL2 or sfml
+    preBuild = ''
+      premake5 gmake2 \
+        --bundled-font="${font}" \
+        --covers=\"${covers_url}\" \
+        --fields=\"${fields_url}\" \
+        --no-core \
+        --no-joystick=false \
+        --pics=\"${pics_url}\" \
+        --prebuilt-core="${ocgcore}/bin" \
+    '';
+
+    enableParallelBuilding = true;
+    NIX_CFLAGS_COMPILE = "-I ${irrlicht-edopro}/include -L ${irrlicht-edopro}/bin";
+    buildFlags = "verbose=true config=release_x64 ygoprodll";
+    makeFlags = "-C build";
+
+    # TODO find out how to get arch from nix for cross compiled
+    installPhase = ''
+      mkdir -p $out/bin
+      cp bin/x64/release/ygoprodll $out/bin
+    '';
+  };
+
+  edopro-script =
+    let
+      assetsToCopy = lib.concatStringsSep "," [
+        #"cacert.pem"
+        "config"
+        "deck"
+        "COPYING.txt"
+        "expansions"
+        "lflists"
+        "notices"
+        "puzzles"
+        "fonts"
+        "script"
+        "skin"
+        "sound"
+        "textures"
+        "WindBot"
+      ];
+    in
+    writeShellScriptBin "EDOPro" ''
+      set -eu
+      EDOPRO_DIR="$HOME/.local/share/edopro"
```

Comment:
We really should support the XDG base directory spec.

Reply:
> We really should support the XDG base directory spec.

I agree, but I did some digging and I couldn't get it to work with xdg. All my env variables were unset, not sure why tbh.

Reply:
Does the snippet I pointed out to you earlier not work? Here's a test case directly from my shell:

```console
tlater ~ $ echo "${XDG_DATA_HOME:-$HOME/.local/share}/edopro"
/home/tlater/.local/share/edopro
tlater ~ $ export XDG_DATA_HOME=/
tlater ~ $ echo "${XDG_DATA_HOME:-$HOME/.local/share}/edopro"
//edopro
```

Note that bash will automatically collapse double `/`.

Reply:
You forgot that ${} is overridden for nix to do its evals, unless theres another way to write it in bash I'm not sure how to.

Reply:
You just escape it with `''`, see: https://nix.dev/manual/nix/2.18/language/values#type-string